packages/eslint-plugin/docs/rules/no-floating-promises.mdx

```diff
@@ -119,6 +119,54 @@ await (async function () {
 })();
 ```
 
+### `allowForKnownSafePromises`
+
+This option allows marking specific types as "safe" to be floating. For example, you may need to do this in the case of libraries whose APIs return Promises whose rejections are safely handled by the library.
+
+This option takes an array of type specifiers to consider safe.
+Each item in the array must have one of the following forms:
+
+- A type defined in a file (`{ from: "file", name: "Foo", path: "src/foo-file.ts" }` with `path` being an optional path relative to the project root directory)
+- A type from the default library (`{ from: "lib", name: "PromiseLike" }`)
+- A type from a package (`{ from: "package", name: "Foo", package: "foo-lib" }`, this also works for types defined in a typings package).
+
+Additionally, a type may be defined just as a simple string, which then matches the type independently of its origin.
+
```

Comment:
We'll eventually have a consolidated place to describe this shared format... but for now, I really don't want to encourage people to do this.

```suggestion
```

https://github.com/typescript-eslint/typescript-eslint/pull/8502#discussion_r1541768874

Reply:
ah, sorry, missed that comment. Will revert.